docs/proposals/en/cache-reuse.md

```diff
@@ -0,0 +1,97 @@
+
+Fluid-based CacheBackend was implemented in the last PR, which provides KubeDL with the basic ability to cache datasets. However, in previous implementation, CacheBackend was bound to the DL job that created it, which meant that if the job was deleted after completed, the CacheBackend attached to the job was also deleted.
+
+To enable the CacheBackend created by the previous Job can be reused by the next Job, we want to keep the CacheBackend even if the job that created it is deleted. In this way, user can avoid creating CacheBackend repeatedly and skip the step of caching remote files, thus improving the efficiency of CacheBackend. 
+
+Here's a simple illustration:
+
+<img src="../../img/cache_reuse.png" style="zoom:20%;" />
+
+In addition, to release CacheBackend that has not been used for a long time, I think we can also add an option to CacheBackend, maxIdleTime, to set the maximum amount of time CacheBackend can be idle. KubeDL automatically deletes the unused CacheBackend to ensure that cache files do not consume too many space in the cluster.
+
+### Goals
+
+- CacheBackend in KubeDL can be retained after a job ends and can be reused by specifying the name of CacheBackend to improve utilization efficiency.
+- Add maxIdleTime options for CacheBackend to configure when CacheBackend can be deleted.
+
+## Use case
+
+Added use cases for cache reuse:
+
+1. The user wants to simply create a one-off CacheBackend that will be deleted after the job completes.
+2. The user creates a new CacheBackend and expects it to be retained after this job completes so that the cached dataset files can be reused in the next job.
+3. The user wants to specify the name of CacheBackend when creating a job to use the CacheBackend retained in a previous job
+4. When creating CacheBackend, the user can specify maxIdleTime. If a CacheBackend is idle for more than maxIdleTime, the CacheBackend is automatically deleted to save space.
+
+### Demo Configuration
+
+```yaml
+apiVersion: "training.kubedl.io/v1alpha1"
+kind: "TFJob"
+metadata:
+  name: "mnist"
+  namespace: kubedl
+spec:
+  cleanPodPolicy: None
+  CacheBackend:
++   name: "DemoCacheBackend"
+    mountPath: "/data"   
+    dataset:
+      dataSources:
+        - location: local:///dataset/mnist
+          subDirName: mnist
+    cacheEngine:
+      fluid:
+        alluxioRuntime:
+          replicas: 1
+          tieredStorage:
+            - cachePath: /dev/shm
+              quota: "1Gi"
+              mediumType: MEM
++   options:
++     maxIdleTime: "10h"
+  tfReplicaSpecs:
+    ...
+    No additional dataset volumes need to be configured. 
+    The cache controller will automatically inject the pvc created by fluid into the container
+```
+
+## Proposal
+
+For CacheBackend in KubeDL to have the ability to be reused by multiple jobs, we may need to add the following fields to existing CacheBackend:
+
+1. `spec.CacheBackendName`：A user specifies which CacheBackend to use by giving a `CacheBackendName`
```

Comment:
Yes, `CacheBackendName` in `spec` is redundant. Do you mean just create a struct and wrap `ObjectMeta` in it?

As I understand it, it could be like this:

```go
type CacheBackendTemplate struct {
    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
    Spec CacheBackendSpec `json:"spec,omitempty"`
}

type PyTorchJobSpec struct {
    ...
    CacheBackend *cachev1alpha1.CacheBackendTemplate `json:"cacheBackend"`
    ...
}
```

And the job YAML may be changed like this:

```yaml
apiVersion: "training.kubedl.io/v1alpha1"
kind: "TFJob"
metadata:
  name: "mnist"
  namespace: kubedl
spec:
  cleanPodPolicy: None
  CacheBackend:
    metadata:
        name: "DemoCacheBackend"
    spec:
        mountPath: "/data"   
        dataset:
...
```

Reply:
yes, that's what I meant, else the `name` in spec will be duplicated with the `name` in metadata

Reply:
OK, I got it